Sources/BSON/ByteBuffer+BSON.swift

```diff
@@ -0,0 +1,48 @@
+import NIO
+
+extension ByteBuffer {
+    @discardableResult
+    internal mutating func writeCString(_ string: String) -> Int {
+        let written = self.writeString(string + "\0")
+        return written
+    }
+
+    internal mutating func readCString() throws -> String {
```

Comment:
it's not clear to me why this is split into 3 methods, do we expect to need to use either of `getCString` or `getBSONKey` out of the context of `readCString`?

Reply:
Yea I should've called this out, these helpers are from the work I've done for completing the document iterator, I wanted a get* version of the readCString so that I have an equivalent method that doesn't move the readerIndex
The BSONKey one I can leave out of this PR and include it later but the thinking there was there is code where I need get the the key value including its null byte for equality reasons and so that I don't have to translate the string back to utf8 just to get an accurate length

Reply:
I think for review purposes it will probably be clearer if we can bring in all these changes incrementally as we use them; otherwise it's hard for reviewers to evaluate the solution in the context it's being used in.
separately, any that are left in should have at least a basic 1-2 sentence docstring added describing their behavior.